Installing on macOS fails due to failed watchdog installation

Due to https://github.com/gorakhargosh/watchdog/issues/689 and `watchdog` not providing built distributions, installation has been failing on macOS:

```
$ python3 -m pip install functions-framework
Collecting functions-framework
  Downloading https://files.pythonhosted.org/packages/63/bc/9ec82d36af2f58ad39522f66684561f316a4fd8133fdf645d6e739025fb3/functions_framework-2.0.0-py3-none-any.whl
Collecting click<8.0,>=7.0 (from functions-framework)
  Using cached https://files.pythonhosted.org/packages/d2/3d/fa76db83bf75c4f8d338c2fd15c8d33fdd7ad23a9b5e57eb6c5de26b430e/click-7.1.2-py2.py3-none-any.whl
Collecting watchdog>=0.10.0 (from functions-framework)
  Downloading https://files.pythonhosted.org/packages/e6/76/39d123d37908a772b6a281d85fbb4384d9db7e13d19d10ad409006bd2962/watchdog-1.0.1.tar.gz (97kB)
     |████████████████████████████████| 102kB 4.1MB/s 
Collecting gunicorn<21.0,>=19.2.0; platform_system != "Windows" (from functions-framework)
  Downloading https://files.pythonhosted.org/packages/69/ca/926f7cd3a2014b16870086b2d0fdc84a9e49473c68a8dff8b57f7c156f43/gunicorn-20.0.4-py2.py3-none-any.whl (77kB)
     |████████████████████████████████| 81kB 14.2MB/s 
Collecting cloudevents<1.0 (from functions-framework)
  Downloading https://files.pythonhosted.org/packages/50/f9/ff5db473ee17d2a83a697ea238e6cf41970057fe082b3b5a5c18abc4511b/cloudevents-0.3.0-py3-none-any.whl
Collecting flask<2.0,>=1.0 (from functions-framework)
  Using cached https://files.pythonhosted.org/packages/f2/28/2a03252dfb9ebf377f40fba6a7841b47083260bf8bd8e737b0c6952df83f/Flask-1.1.2-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=3.0 in /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/site-packages (from gunicorn<21.0,>=19.2.0; platform_system != "Windows"->functions-framework) (41.2.0)
Collecting Werkzeug>=0.15 (from flask<2.0,>=1.0->functions-framework)
  Using cached https://files.pythonhosted.org/packages/cc/94/5f7079a0e00bd6863ef8f1da638721e9da21e5bacee597595b318f71d62e/Werkzeug-1.0.1-py2.py3-none-any.whl
Collecting itsdangerous>=0.24 (from flask<2.0,>=1.0->functions-framework)
  Using cached https://files.pythonhosted.org/packages/76/ae/44b03b253d6fade317f32c24d100b3b35c2239807046a4c953c7b89fa49e/itsdangerous-1.1.0-py2.py3-none-any.whl
Collecting Jinja2>=2.10.1 (from flask<2.0,>=1.0->functions-framework)
  Using cached https://files.pythonhosted.org/packages/30/9e/f663a2aa66a09d838042ae1a2c5659828bb9b41ea3a6efa20a20fd92b121/Jinja2-2.11.2-py2.py3-none-any.whl
Collecting MarkupSafe>=0.23 (from Jinja2>=2.10.1->flask<2.0,>=1.0->functions-framework)
  Downloading https://files.pythonhosted.org/packages/0c/12/37f68957526d1ec0883b521934b4e1b8ff3dd8e4fab858a5bf3e487bcee9/MarkupSafe-1.1.1-cp38-cp38-macosx_10_9_x86_64.whl
Building wheels for collected packages: watchdog
  Building wheel for watchdog (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Library/Developer/CommandLineTools/usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/f3/n1z3ncc57rzcsdmjrdqpk33000rnw4/T/pip-install-2va8a80x/watchdog/setup.py'"'"'; __file__='"'"'/private/var/folders/f3/n1z3ncc57rzcsdmjrdqpk33000rnw4/T/pip-install-2va8a80x/watchdog/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/f3/n1z3ncc57rzcsdmjrdqpk33000rnw4/T/pip-wheel-7b2gmfbc --python-tag cp38
       cwd: /private/var/folders/f3/n1z3ncc57rzcsdmjrdqpk33000rnw4/T/pip-install-2va8a80x/watchdog/
  Complete output (146 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.14.6-x86_64-3.8
  creating build/lib.macosx-10.14.6-x86_64-3.8/watchdog
  copying src/watchdog/watchmedo.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog
  copying src/watchdog/version.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog
  copying src/watchdog/events.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog
  copying src/watchdog/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog
  creating build/lib.macosx-10.14.6-x86_64-3.8/watchdog/utils
  copying src/watchdog/utils/patterns.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/utils
  copying src/watchdog/utils/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/utils
  copying src/watchdog/utils/dirsnapshot.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/utils
  copying src/watchdog/utils/delayed_queue.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/utils
  copying src/watchdog/utils/platform.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/utils
  copying src/watchdog/utils/bricks.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/utils
  copying src/watchdog/utils/echo.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/utils
  creating build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/fsevents.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/inotify.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/api.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/inotify_buffer.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/winapi.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/read_directory_changes.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/kqueue.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/inotify_c.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/polling.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  copying src/watchdog/observers/fsevents2.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/observers
  creating build/lib.macosx-10.14.6-x86_64-3.8/watchdog/tricks
  copying src/watchdog/tricks/__init__.py -> build/lib.macosx-10.14.6-x86_64-3.8/watchdog/tricks
  running egg_info
  writing src/watchdog.egg-info/PKG-INFO
  writing dependency_links to src/watchdog.egg-info/dependency_links.txt
  writing entry points to src/watchdog.egg-info/entry_points.txt
  writing requirements to src/watchdog.egg-info/requires.txt
  writing top-level names to src/watchdog.egg-info/top_level.txt
  reading manifest file 'src/watchdog.egg-info/SOURCES.txt'
  reading manifest template 'MANIFEST.in'
  warning: no files found matching '*.h' under directory 'src'
  writing manifest file 'src/watchdog.egg-info/SOURCES.txt'
  running build_ext
  building '_watchdog_fsevents' extension
  creating build/temp.macosx-10.14.6-x86_64-3.8
  creating build/temp.macosx-10.14.6-x86_64-3.8/src
  clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -iwithsysroot/System/Library/Frameworks/System.framework/PrivateHeaders -iwithsysroot/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/Headers -arch arm64 -arch x86_64 -DWATCHDOG_VERSION_STRING="1.0.1" -DWATCHDOG_VERSION_MAJOR=1 -DWATCHDOG_VERSION_MINOR=0 -DWATCHDOG_VERSION_BUILD=1 -I/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8 -c src/watchdog_fsevents.c -o build/temp.macosx-10.14.6-x86_64-3.8/src/watchdog_fsevents.o -std=c99 -pedantic -Wall -Wextra -fPIC -Wno-nullability-completeness -Wno-nullability-extension -Wno-newline-eof -Wno-error=unused-command-line-argument
  In file included from src/watchdog_fsevents.c:22:
  In file included from /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8/Python.h:11:
  In file included from /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0/include/limits.h:21:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/limits.h:63:
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/cdefs.h:807:2: error: Unsupported architecture
  #error Unsupported architecture
   ^
  In file included from src/watchdog_fsevents.c:22:
  In file included from /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8/Python.h:11:
  In file included from /Library/Developer/CommandLineTools/usr/lib/clang/12.0.0/include/limits.h:21:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/limits.h:64:
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/machine/limits.h:8:2: error: architecture not supported
  #error architecture not supported
   ^
  In file included from src/watchdog_fsevents.c:22:
  In file included from /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8/Python.h:25:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:64:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/_stdio.h:71:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/_types.h:27:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:33:
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/machine/_types.h:34:2: error: architecture not supported
  #error architecture not supported
   ^
  In file included from src/watchdog_fsevents.c:22:
  In file included from /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8/Python.h:25:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:64:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/_stdio.h:71:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/_types.h:27:
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:55:9: error: unknown type name '__int64_t'
  typedef __int64_t       __darwin_blkcnt_t;      /* total blocks */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:56:9: error: unknown type name '__int32_t'; did you mean '__int128_t'?
  typedef __int32_t       __darwin_blksize_t;     /* preferred block size */
          ^
  note: '__int128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:57:9: error: unknown type name '__int32_t'; did you mean '__int128_t'?
  typedef __int32_t       __darwin_dev_t;         /* dev_t */
          ^
  note: '__int128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:60:9: error: unknown type name '__uint32_t'; did you mean '__uint128_t'?
  typedef __uint32_t      __darwin_gid_t;         /* [???] process and group IDs */
          ^
  note: '__uint128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:61:9: error: unknown type name '__uint32_t'; did you mean '__uint128_t'?
  typedef __uint32_t      __darwin_id_t;          /* [XSI] pid_t, uid_t, or gid_t*/
          ^
  note: '__uint128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:62:9: error: unknown type name '__uint64_t'
  typedef __uint64_t      __darwin_ino64_t;       /* [???] Used for 64 bit inodes */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:68:9: error: unknown type name '__darwin_natural_t'
  typedef __darwin_natural_t __darwin_mach_port_name_t; /* Used by mach */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:70:9: error: unknown type name '__uint16_t'; did you mean '__uint128_t'?
  typedef __uint16_t      __darwin_mode_t;        /* [???] Some file attributes */
          ^
  note: '__uint128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:71:9: error: unknown type name '__int64_t'
  typedef __int64_t       __darwin_off_t;         /* [???] Used for file sizes */
          ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:72:9: error: unknown type name '__int32_t'; did you mean '__int128_t'?
  typedef __int32_t       __darwin_pid_t;         /* [???] process and group IDs */
          ^
  note: '__int128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:73:9: error: unknown type name '__uint32_t'; did you mean '__uint128_t'?
  typedef __uint32_t      __darwin_sigset_t;      /* [???] signal set */
          ^
  note: '__uint128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:74:9: error: unknown type name '__int32_t'; did you mean '__int128_t'?
  typedef __int32_t       __darwin_suseconds_t;   /* [???] microseconds */
          ^
  note: '__int128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:75:9: error: unknown type name '__uint32_t'; did you mean '__uint128_t'?
  typedef __uint32_t      __darwin_uid_t;         /* [???] user IDs */
          ^
  note: '__uint128_t' declared here
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types.h:76:9: error: unknown type name '__uint32_t'; did you mean '__uint128_t'?
  typedef __uint32_t      __darwin_useconds_t;    /* [???] microseconds */
          ^
  note: '__uint128_t' declared here
  In file included from src/watchdog_fsevents.c:22:
  In file included from /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8/Python.h:25:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:64:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/_stdio.h:71:
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/_types.h:43:9: error: unknown type name '__uint32_t'; did you mean '__uint128_t'?
  typedef __uint32_t      __darwin_wctype_t;
          ^
  note: '__uint128_t' declared here
  In file included from src/watchdog_fsevents.c:22:
  In file included from /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/include/python3.8/Python.h:25:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/stdio.h:64:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/_stdio.h:75:
  In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/sys/_types/_va_list.h:31:
  /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include/machine/types.h:37:2: error: architecture not supported
  #error architecture not supported
   ^
  fatal error: too many errors emitted, stopping now [-ferror-limit=]
  20 errors generated.
  error: command 'clang' failed with exit status 1
  ----------------------------------------
  ERROR: Failed building wheel for watchdog
  Running setup.py clean for watchdog
Failed to build watchdog
Installing collected packages: click, watchdog, gunicorn, cloudevents, Werkzeug, itsdangerous, MarkupSafe, Jinja2, flask, functions-framework
ERROR: Could not install packages due to an EnvironmentError: [Errno 13] Permission denied: '/Library/Python/3.8'
Consider using the `--user` option or check the permissions.

WARNING: You are using pip version 19.2.3, however version 20.3.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

--- a/setup.py
+++ b/setup.py
@@ -51,7 +51,7 @@ setup(
     install_requires=[
         "flask>=1.0,<2.0",
         "click>=7.0,<8.0",
-        "watchdog>=0.10.0",
+        "watchdog>=1.0.0,<2.0.0",
         "gunicorn>=19.2.0,<21.0; platform_system!='Windows'",
         "cloudevents>=1.2.0,<2.0.0",
     ],

